src/Adapter/PDF/DeliverySlipPdfGenerator.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\PDF;
+
+use Context;
+use Order;
+use PDF;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\PDF\PDFGeneratorInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+use Validate;
+
+/**
+ * Generates delivery slip for given order
+ *
+ * @internal
+ */
+final class DeliverySlipPdfGenerator implements PDFGeneratorInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generatePDF(array $orderId)
```

Comment:
I didn't want to introduce new interface for generating PDFs, so I've used existing one. However, I've used it in a way it wasn't meant to be implemented. :confused: You can take a look at `OrderController` to see how it's called.

Wdyt about it?

Reply:
> So I'm wondering whether we should deprecate it.

This is a big problem imo. We cannot introduce new interface in 1.7.5 and then deprecate in 1.7.7, it's just wrong and I think you understand it. :/

How could we solve it? It's a recurrent issue we face.

Maybe each time we introduce new component, it should be done in separate PR, so everyone can discuss it and provide feedback, otherwise this kind of interfaces blurs in context of migration PRs and gets merged without being well designed. :/

Reply:
> > So I'm wondering whether we should deprecate it.
> 
> This is a big problem imo. We cannot introduce new interface in 1.7.5 and then deprecate in 1.7.7, it's just wrong and I think you understand it. :/

I dont think @mickaelandrieu would be against it 😛

> Maybe each time we introduce new component, it should be done in separate PR, so everyone can discuss it and provide feedback, otherwise this kind of interfaces blurs in context of migration PRs and gets merged without being well designed. :/

Yes indeed

Reply:
Since this could be considered as a separate feature, I suggest improving it with new implementation in separate PR, wdyt?

Reply:
Agreed 👍

Reply:
Added to issue todo-list ✅